src/tools/ceph_conf.cc

```diff
@@ -163,8 +167,54 @@ static int dump_all(const string& format)
   }
 }
 
+static int _parse_argv(int argc, const char **argv, vector<const char*>& args,
+                       std::string& new_name_value)
+{
+  int next = 0;
+  argv_to_vec(argc, argv, args);
+
+  for (const auto& arg : args) {
+    next++;
+    if (!strcmp(arg, "--name") || !strcmp(arg, "-n")) {
```

Comment:
```c++
auto [type, id] = [&] -> std::pair<string, string> {
  for (auto i = args.begin(); i != args.end(); ) {
    string name;
    if (ceph_argparse_witharg(args, i, &name, "--name", "-n", (char*)NULL)) {
      size_t dot_pos = name.rfind('.');
      if (dot_pos == name.npos) {
        return {name, ""};
      } else {
        return {name.substr(0, dot_pos), name.substr(dot_pos + 1)};
      }
    } else {
      ++i;
    }
  }
}
if (type.empty() || id.empty()) {
  return;
}
static const char* daemon_types[] = {"mon", "osd", "mds", "mgr"};
if (std::find(std::begin(daemon_types), std::end(daemon_types), type) != 
    std::end(daemon_types)) {
  // only override name and pid for non-daemon names
  return;
}
try {
  int pid = std::stoi(id);
} catch (const std::logic_error&) {
  // only override pid for $id which looks like pid
  return;
}
args.push_back("--name");
args.push_back(name);
setenv("PID", id().c_str(), 1);
```

Reply:
@lxbsz hi Xiubo, i am proposing a change like this for better readability. could you please take a look ?

Reply:
> three things,
> 
> 1. i think we need to either document this function or use a better name for this function. as this change is a little bit hackish.
Yeah, sounds good and I will doc it.

> 2. i'd suggest use `ceph_argparse_witharg()` instead of updating the "name" element in-place.
Will check and try it.

> 3. please stick to "$name.$id". no need to lower the bar by searching all elements after splitting the "name" string, just check the first one would suffice.
Okay.

Reply:
> @lxbsz hi Xiubo, i am proposing a change like this for better readability. could you please take a look ?

Sure, I will do it today. Thanks @tchaikov